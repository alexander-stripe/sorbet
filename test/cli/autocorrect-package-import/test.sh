#!/usr/bin/env bash

cwd="$(pwd)"
tmp="$(mktemp -d)"

mkdir -p "${tmp}"/{a,b}
cp "$cwd/test/cli/autocorrect-package-import/a/__package.rb" "$tmp/a"
cp "$cwd/test/cli/autocorrect-package-import/a/foo.rb" "$tmp/a"
cp "$cwd/test/cli/autocorrect-package-import/b/__package.rb" "$tmp/b"
cp "$cwd/test/cli/autocorrect-package-import/b/foo.rb" "$tmp/b"

cd "$tmp" || exit 1

if "$cwd/main/sorbet" --silence-dev-message --stripe-packages -a . 2>&1; then
    echo "Expected to fail!"
    exit 1
fi

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make sure that the autocorrect applied
cat a/__package.rb

rm autocorrect-abstract.rb
