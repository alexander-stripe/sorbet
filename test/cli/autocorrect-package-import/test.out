a/foo.rb:4: `B::Foo` resolves but its package is not imported https://srb.help/3718
     4 |  B::Foo.new
          ^^^^^^
    b/__package.rb:3: Exported from package here
     3 |class B < PackageSpec
        ^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Done
    a/__package.rb:4: Inserted `import B`
     4 |class A < PackageSpec
                             ^
Errors: 1

--------------------------------------------------------------------------

# typed: strict
# enable-packager: true

class A < PackageSpec
  import B
end
